htdocs/compta/facture/class/facture.class.php

```diff
@@ -6721,6 +6735,291 @@ public function update_total()
 		}
 	}
 
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+	/**
+	 *	Update total_ht, total_ttc, total_vat, total_localtax1, total_localtax2 for an object (sum of lines).
+	 *  Must be called at end of methods addline or updateline.
+	 *
+	 *	@param	int		$exclspec          	>0 = Exclude special product (product_type=9)
+	 *  @param  string	$roundingadjust    	'none'=Do nothing, 'auto'=Use default method (MAIN_ROUNDOFTOTAL_NOT_TOTALOFROUND if defined, or '0'), '0'=Force mode Total of rounding, '1'=Force mode Rounding of total
+	 *  @param	int		$nodatabaseupdate	1=Do not update database total fields of the main object. Update only properties in memory. Can be used to save SQL when this method is called several times, so we can do it only once at end.
+	 *  @param	Societe	$seller				If roundingadjust is '0' or '1' or maybe 'auto', it means we recalculate total for lines before calculating total for object and for this, we need seller object (used to analyze lines to check corrupted data).
+	 *	@return	int    			           	Return integer <0 if KO, >0 if OK
+	 */
+	public function update_price($exclspec = 0, $roundingadjust = 'none', $nodatabaseupdate = 0, $seller = null)
+	{
+		// phpcs:enable
+		global $conf, $hookmanager, $action;
+
+		$parameters = array('exclspec' => $exclspec, 'roundingadjust' => $roundingadjust, 'nodatabaseupdate' => $nodatabaseupdate, 'seller' => $seller);
+		$reshook = $hookmanager->executeHooks('updateTotalPrice', $parameters, $this, $action); // Note that $action and $object may have been modified by some hooks
+		if ($reshook > 0) {
+			return 1; // replacement code
+		} elseif ($reshook < 0) {
+			return -1; // failure
+		} // reshook = 0 => execute normal code
+
+		// Some external module want no update price after a trigger because they have another method to calculate the total (ex: with an extrafield)
+		$MODULE = "";
+		if ($this->element == 'propal') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_PROPOSAL";
+		} elseif ($this->element == 'commande' || $this->element == 'order') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_ORDER";
+		} elseif ($this->element == 'facture' || $this->element == 'invoice') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_INVOICE";
+		} elseif ($this->element == 'facture_fourn' || $this->element == 'supplier_invoice' || $this->element == 'invoice_supplier' || $this->element == 'invoice_supplier_rec') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_INVOICE";
+		} elseif ($this->element == 'order_supplier' || $this->element == 'supplier_order') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_ORDER";
+		} elseif ($this->element == 'supplier_proposal') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_PROPOSAL";
+		}
+
+		if (!empty($MODULE)) {
+			if (!empty($conf->global->$MODULE)) {
```

Comment:
Not compatible with recent php version.
Can you replace with
if (getDolGlobalString($MODULE))

Ps: as this is addition of a new function, why not having submitting it into a separated PR. Splitting large PR into several ones increase dramatically speed and ease of integration.

Reply:
Thanks @eldy. Fix pushed